cmds/db-manager/main.go

```diff
@@ -0,0 +1,228 @@
+}
+
+// Direction is an alias for int indicating the direction and steps of migration
+type Direction int
+
+func (d Direction) String() string {
+	if d > 0 {
+		return "Up"
+	} else if d < 0 {
+		return "Down"
+	}
+	return "No Change"
+}
+
+var (
+	path      = flag.String("schemas_dir", "/Users/pcharlie/charlie_dss/dss/build/db-schemas/scd", "path to db migration files directory")
+	dbVersion = flag.String("db_version", "", "the db version to migrate to (ex: v1.0.0)")
+	step      = flag.Int("migration_step", 1, "the db migration step to go to")
+
+	cockroachParams = struct {
+		host    *string
+		port    *int
+		sslMode *string
+		sslDir  *string
+		user    *string
+	}{
+		host:    flag.String("cockroach_host", "0.0.0.0", "cockroach host to connect to"),
+		port:    flag.Int("cockroach_port", 26257, "cockroach port to connect to"),
+		sslMode: flag.String("cockroach_ssl_mode", "disable", "cockroach sslmode"),
+		user:    flag.String("cockroach_user", "root", "cockroach user to authenticate as"),
+		sslDir:  flag.String("cockroach_ssl_dir", "", "directory to ssl certificates. Must contain files: ca.crt, client.<user>.crt, client.<user>.key"),
+	}
+)
+
+func main() {
+	flag.Parse()
+	if *path == "" {
+		log.Panic("Must specify schemas_dir path")
+	}
+	if (*dbVersion == "" && *step == 0) || (*dbVersion != "" && *step != 0) {
+		log.Panic("Must specify a db_version or migration_step to goto")
+	}
+	uriParams := map[string]string{
+		"host":             *cockroachParams.host,
+		"port":             strconv.Itoa(*cockroachParams.port),
+		"user":             *cockroachParams.user,
+		"ssl_mode":         *cockroachParams.sslMode,
+		"ssl_dir":          *cockroachParams.sslDir,
+		"application_name": "SchemaManager",
+		"db_name":          filepath.Base(*path),
+	}
+	postgresURI, err := dssCockroach.BuildURI(uriParams)
+	if err != nil {
+		log.Panic("Failed to build URI", zap.Error(err))
+	}
+	crdbURI := strings.Replace(postgresURI, "postgresql", "cockroachdb", 1)
+	*path = fmt.Sprintf("file://%v", *path)
+	migrater, migraterErr := migrate.New(*path, crdbURI)
+	if migraterErr != nil {
+		log.Panic(migraterErr)
+	}
+	myMigrater := &MyMigrate{migrater}
+	defer func() {
+		if migraterErr == nil {
+			if _, err := myMigrater.Close(); err != nil {
+				log.Println(err)
+			}
+		}
+	}()
+	if migraterErr == nil {
+		myMigrater.PrefetchMigrations = 10
+		myMigrater.LockTimeout = time.Duration(15) * time.Second
+
+		// handle Ctrl+c
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, syscall.SIGINT)
+		go func() {
+			for range signals {
+				log.Println("Stopping after this running migration ...")
+				myMigrater.GracefulStop <- true
+				return
+			}
+		}()
+	}
+	totalMove := 0
+	migrateDirection := myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	preMigrationStep, _, _ := myMigrater.Version()
+	for migrateDirection != 0 {
+		migraterErr = myMigrater.Steps(int(migrateDirection))
+		if migraterErr != nil {
+			log.Fatal(migraterErr)
```

Comment:
I think Fatal calls os.Exit without running defers, so do we want log.Panic here to ensure myMigrater is Closed in the above defer?